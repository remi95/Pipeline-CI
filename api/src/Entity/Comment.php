<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity(repositoryClass="App\Repository\CommentRepository")
 * @ORM\HasLifecycleCallbacks
 */
class Comment
{
    /**
     * @ORM\Id()
     * @ORM\GeneratedValue()
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="text")
     */
    private $content;

    /**
     * @ORM\ManyToOne(targetEntity="App\Entity\User", inversedBy="comments")
     * @ORM\JoinColumn(nullable=false)
     */
    private $user;

    /**
     * @ORM\ManyToOne(targetEntity="App\Entity\Request", inversedBy="comments")
     */
    private $request;

    /**
     * @ORM\ManyToOne(targetEntity="App\Entity\Favor", inversedBy="comments")
     */
    private $favor;

    /**
     * @ORM\Column(type="datetime")
     */
    private $createdAt;

    /**
     * @ORM\Column(type="datetime")
     */
    private $updatedAt;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getContent(): ?string
    {
        return $this->content;
    }

    public function setContent(string $content): self
    {
        $this->content = $content;

        return $this;
    }

    public function getUser(): ?User
    {
        return $this->user;
    }

    public function setUser(?User $user): self
    {
        $this->user = $user;

        return $this;
    }

    public function getRequest(): ?Request
    {
        return $this->request;
    }

    public function setRequest(?Request $request): self
    {
        $this->request = $request;

        return $this;
    }

    public function getFavor(): ?Favor
    {
        return $this->favor;
    }

    public function setFavor(?Favor $favor): self
    {
        $this->favor = $favor;

        return $this;
    }

    public function __toString()
    {
        return substr($this->getContent(), 0, 30)."...";
    }

    public function getCreatedAt(): ?\DateTimeInterface
    {
        return $this->createdAt;
    }

    public function setCreatedAt(\DateTimeInterface $createdAt): self
    {
        $this->createdAt = $createdAt;

        return $this;
    }

    public function getUpdatedAt(): ?\DateTimeInterface
    {
        return $this->updatedAt;
    }

    public function setUpdatedAt(\DateTimeInterface $updatedAt): self
    {
        $this->updatedAt = $updatedAt;

        return $this;
    }

    /**
     * @ORM\PrePersist
     * @ORM\PreUpdate
     */
    public function updatedTimestamps(): void
    {
        $this->setUpdatedAt(new \DateTime('now'));
        if ($this->getCreatedAt() === null) {
            $this->setCreatedAt(new \DateTime('now'));
        }
    }
}
