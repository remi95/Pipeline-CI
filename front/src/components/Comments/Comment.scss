.comments {
  textarea {
    width: 100%;
  }
}

.comment {
  padding: 1.5rem 0;

  &:not(:last-child) {
    border-bottom: 1px solid lightgrey;
  }

  img {
    width: 35px;
    height: 35px;
    border-radius: 100%;
    margin: 1rem;
  }

  &-content {
    margin-left: calc(35px + 2rem);
  }
}
