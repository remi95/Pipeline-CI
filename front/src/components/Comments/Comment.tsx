import React from 'react';
import dayjs from 'dayjs';
import { Comment as CommentModel } from '../../models/CommentModel';
import './Comment.scss';
import UserPlaceholder from '../../assets/images/user.png';
import { BASE_URL } from '../../config/api';

type Props = {
  comment: CommentModel;
};

const Comment: React.FC<Props> = (props: Props) => {
  const { comment } = props;

  return (
    <div className="comment">
      <div className="d-flex justify-content-between align-items-center">
        <div>
          <img
            src={comment.user.picture ? BASE_URL + comment.user.picture.path : UserPlaceholder}
            alt={`${comment.user.firstname}_profil`}
          />
          <span className="font-weight-bold">{`${comment.user.firstname} ${comment.user.lastname}`}</span>
        </div>

        <span className="float-right">{dayjs(comment.createdAt).format('Le DD MMMM YYYY Ã  HH:mm')}</span>
      </div>
      <div className="comment-content">
        {comment.content}
      </div>
    </div>
  );
};

export default Comment;
