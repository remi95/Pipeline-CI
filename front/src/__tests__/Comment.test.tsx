import React from 'react';
import renderer from 'react-test-renderer';
import Comment from '../components/Comments/Comment';

const commentMock = {
  id: 1,
  content: 'Test commentaire',
  createdAt: '2020-04-21T00:00:00',
  user: {
    id: 1,
    email: 'simon.toulouze@gmail.com',
    password: 'JjjjHhdazbaNHDZNUdnzua99!',
    lastname: 'Toulouze',
    firstname: 'Simon',
    birthdate: '1993-08-20T17:00:00+01:00',
    phone: '+33611846901',
    picture: {
      id: 2,
      name: '77hhhHGgeg277e6dgJkNggdqvVVd9.jpg',
      originalName: 'jesuisunephoto.jpg',
      path: 'https://images.pexels.com/photos/207962',
    },
    badges: [],
    requests: [],
    favors: [],
  },
};

describe('Comment component tests', () => {
  test('Should render a correct Comment snapshot', () => {
    const component = renderer.create(<Comment comment={commentMock} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
