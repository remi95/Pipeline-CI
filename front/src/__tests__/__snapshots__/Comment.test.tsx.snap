// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment component tests Should render a correct Comment snapshot 1`] = `
<div
  className="comment"
>
  <div
    className="d-flex justify-content-between align-items-center"
  >
    <div>
      <img
        alt="Simon_profil"
        src="http://localhosthttps://images.pexels.com/photos/207962"
      />
      <span
        className="font-weight-bold"
      >
        Simon Toulouze
      </span>
    </div>
    <span
      className="float-right"
    >
      Le 21 April 2020 Ã  00:00
    </span>
  </div>
  <div
    className="comment-content"
  >
    Test commentaire
  </div>
</div>
`;
